server/src/main/java/org/tctalent/server/service/db/CandidateExamService.java

```diff
@@ -45,8 +47,15 @@ CandidateExam createExam(
      * @param request Request containing exam updates
      * @return List of updated candidate exam records
      */
+    @NotNull
     CandidateExam updateCandidateExam(UpdateCandidateExamRequest request);
 
+    /**
+     * List of CandidateExam associated to candidate with given candidate id.
+     * @param id ID of candidate whose CandidateExam we want
+     * @return list of candidate CandidateExam
```

Comment:
Good to document the behaviour if the candidate id is not valid. Some methods will throw an Exception but I note that this method will just not return any exams. That is OK - but it should be documented

```suggestion
     * @return list of candidate CandidateExam - empty if the candidate id is unknown
```